client/starwhale/api/_impl/dataset/builder.py

```diff
@@ -48,10 +43,212 @@
 _header_size = _header_struct.size
 _header_version = 0
 
-
 _BDType = t.TypeVar("_BDType", bound="BaseBuildExecutor")
 
 
+class BinWriter(Protocol):
+    total_bin_size: int
+
+    def write_row(self, row: TabularDatasetRow) -> None:
+        """
+        Find large bytes or local fs file in row data. Convert them to accessible link
+        """
+        ...
+
+    def flush(self) -> None:
+        ...
+
+    def __enter__(self) -> BinWriter:
+        return self
+
+    def __exit__(
+        self,
+        type: t.Optional[t.Type[BaseException]],
+        value: t.Optional[BaseException],
+        trace: TracebackType,
+    ) -> None:
+        ...
+
+
+class FragmentBinWriter:
```

Comment:
FragmentBinWriter -> SWDSBinWriter ?

reason: the bin format is swds unique format.

Reply:
> FragmentBinWriter -> SWDSBinWriter ?
> 
> reason: the bin format is swds unique format.

done